flavors/grapheneos/default.nix

```
@@ -10,105 +10,153 @@ let
-  source.dirs."kernel/google/sunfish".enable = false;
-  source.dirs."kernel/google/redbull".enable = false;
-  source.dirs."kernel/google/barbet".enable = false;
-
-  kernel.enable = mkDefault (elem config.deviceFamily phoneDeviceFamilies);
-
-  # Enable Vanadium (GraphaneOS's chromium fork).
-  apps.vanadium.enable = mkDefault true;
-  webview.vanadium.enable = mkDefault true;
-  webview.vanadium.availableByDefault = mkDefault true;
-
-  apps.seedvault.includedInFlavor = mkDefault true;
-  apps.updater.includedInFlavor = mkDefault true;
-
-  # Remove upstream prebuilt versions from build. We build from source ourselves.
-  removedProductPackages = [ "TrichromeWebView" "TrichromeChrome" "webview" ];
-  source.dirs."external/vanadium".enable = false;
-
-  # Override included android-prepare-vendor, with the exact version from
-  # GrapheneOS. Unfortunately, Doing it this way means we don't cache apv
-  # output across vanilla/grapheneos, even if they are otherwise identical.
-  source.dirs."vendor/android-prepare-vendor".enable = false;
-  nixpkgs.overlays = [ (self: super: {
-    android-prepare-vendor = super.android-prepare-vendor.overrideAttrs (_: {
-      src = config.source.dirs."vendor/android-prepare-vendor".src;
-      patches = [
-        ./apv/0001-Just-write-proprietary-blobs.txt-to-current-dir.patch
-        ./apv/0002-Allow-for-externally-set-config-file.patch
-        ./apv/0003-Add-option-to-use-externally-provided-carrier_list.p.patch
-      ];
-      passthru.evalTimeSrc = builtins.fetchTarball {
-        url = "https://github.com/GrapheneOS/android-prepare-vendor/archive/${config.source.dirs."vendor/android-prepare-vendor".rev}.tar.gz";
-        inherit (config.source.dirs."vendor/android-prepare-vendor") sha256;
+  kernelPrefix = if config.androidVersion >= 13 then "kernel/android" else "kernel/google";
+
+  kernelRepoName = {
+    "sargo" = "crosshatch";
+    "bonito" = "crosshatch";
+    "flame" = "coral";
+    "sunfish" = "coral";
+    "bramble" = "redbull";
+    "redfin" = "redbull";
+    "oriole" = "raviole";
+    "raven" = "raviole";
+    "bluejay" = "bluejay";
+    "panther" = "pantah";
+    "cheetah" = "pantah";
+  }.${config.device} or config.deviceFamily;
+  kernelSourceRelpath = "${kernelPrefix}/${kernelRepoName}";
+  kernelSources = lib.mapAttrs'
+    (path: src: {
+      name = "${kernelSourceRelpath}/${path}";
+      value = src // {
+        enable = false;
       };
-    });
-  }) ];
-
-  # GrapheneOS just disables apex updating wholesale
-  signing.apex.enable = false;
-
-  # Extra packages that should use releasekey
-  signing.signTargetFilesArgs = [ "--extra_apks OsuLogin.apk,ServiceWifiResources.apk=$KEYSDIR/${config.device}/releasekey" ];
-
-  # Leave the existing auditor in the build--just in case the user wants to
-  # audit devices running the official upstream build
-}
+    })
+    (lib.importJSON (./kernel-repos/repo- + "${config.deviceFamily}-${grapheneOSRelease}.json"));
```

```suggestion
    (lib.importJSON (./kernel-repos/repo- + "${kernelRepoName}-${grapheneOSRelease}.json"));
```

I think. Not sure about that though.